OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/AchievementControllerTest.cs

```diff
@@ -31,6 +36,139 @@ public void Setup()
         controller = new AchievementController(achievementService.Object, providerService.Object, providerAdminService.Object, workshopService.Object);
     }
 
+    [Test]
+    public async Task CreateAchievement_WhenAchievementDtoIsNull_ShouldReturnBadRequest()
+    {
+        // Act
+        var result = await controller.Create(null).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<BadRequestObjectResult>(result);
+    }
+
+    [Test]
+    public async Task CreateAchievement_WhenWorkshopIdIsNotValid_ShouldReturnNotFound()
+    {
+        // Arrange
+        var dto = new AchievementCreateDTO();
+        workshopService.Setup(s => s.Exists(It.IsAny<Guid>())).ReturnsAsync(false);
+
+        // Act
+        var result = await controller.Create(dto).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<NotFoundObjectResult>(result);
+    }
+
+    [Test]
+    public async Task CreateAchievement_WhenProviderIsBlocked_ShouldReturnForbidden()
+    {
+        // Arrange
+        var dto = new AchievementCreateDTO();
+        workshopService.Setup(s => s.Exists(It.IsAny<Guid>())).ReturnsAsync(true);
```

Comment:
(Test code clarity) Please consider minimizing usage of `It.IsAny<Guid>()`. Because this way it's not clear that we are checking for specific workshop/provider id. Related to this and subsequent tests.

Reply:
Do we have to use the approach with concrete IDs when we test controllers?

Reply:
Thanks for your explanation. I replaced IDs in the tests.